packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -647,13 +647,13 @@ export class DataStores implements IDisposable {
     }
 
     /**
-     * Called during GC to retrieve the package path of a data store node with the given path.
+     * Called by GC to retrieve the package path of a data store node with the given path.
      */
-    public getDataStorePackagePath(nodePath: string): readonly string[] | undefined {
-        // If the node belongs to a data store, return its package path if the data store is loaded. For DDSs, we return
-        // the package path of the data store that contains it.
+    public async getDataStorePackagePath(nodePath: string): Promise<readonly string[] | undefined> {
+        // If the node belongs to a data store, return its package path. For DDSs, we return the package path of the
+        // data store that contains it.
         const context = this.contexts.get(nodePath.split("/")[1]);
-        return context?.isLoaded ? context.packagePath : undefined;
+        return (await context?.getInitialSnapshotDetails())?.pkg;
```

Comment:
What will the impact of this await be in terms of latency?  Other perf concerns from fetching this?  Does awaiting this trigger some work to be done (I know we have a LazyPromise or something)?

Reply:
It downloads the attributes blob if it has not already been downloaded. However, this blob is already downloaded for all data stores when GC calls `dataStoreContext.isRoot` when GC runs. So this does not have any impact at all.

Reply:
Well, pkg won't be undefined. But the context itself can be undefined if this is requested for a non-data store object.